igvc_platform/src/imu/main.cpp

```diff
@@ -71,32 +62,58 @@ int main(int argc, char** argv)
     ROS_ERROR_STREAM("failed to send message");
   }
 
-  freespace_message response;
-  ret = freespace_readMessage(ids[0], &response, 1000);
-  if (ret != FREESPACE_SUCCESS)
-  {
-    if (ret == FREESPACE_ERROR_TIMEOUT)
-    {
-      ROS_ERROR_STREAM("failed to read IMU TIMEOUT");
-    }
-    else
-    {
-      ROS_ERROR_STREAM("failed to read IMU" << ret);
-    }
-  }
-  if (response.messageType == FREESPACE_MESSAGE_DATAMODECONTROLV2RESPONSE)
-  {
-    ROS_INFO_STREAM("got response" << response.dataModeControlV2Response.modeActual);
+  return ret;
+}
+
+int main(int argc, char** argv)
+{
+  ros::init(argc, argv, "imu");
+
+  ros::NodeHandle nh;
+  ros::NodeHandle pNh("~");
+
+  // Get matrix to correct IMU orientation.
+  std::vector<double> imu_orientation_correction;
```

Comment:
we do not check if param exists and warns. There is a header in utils that will handle this inline.

Reply:
Done